### PR TITLE
tests: Make tests run faster by using Fixture builders

### LINKED ISSUE #310: investigate tests running incredibly slow on mac os

### PR DESCRIPTION
**Pull request by fitzgen**
fixes #310 

Instead of copying directories, just build the fixtures directly in the temporary directory.

-------------------------

Make sure these boxes are checked! 📦✅

- [x] You have the latest version of `rustfmt` installed and have your 
      cloned directory set to nightly
```bash
$ rustup override set nightly
$ rustup component add rustfmt-preview --toolchain nightly
```
- [x] You ran `rustfmt` on the code base before submitting
- [x] You reference which issue is being closed in the PR text

✨✨ 😄 Thanks so much for contributing to wasm-pack! 😄 ✨✨

### CODE REVIEW

File: tests/all/utils/fixture.rs [hunk 1]
```diff
@@ -17,66 +17,123 @@ pub struct Fixture {
...
-    );
+impl Fixture {
+    /// Create a new test fixture in a temporary directory.
+    pub fn new() -> Fixture {
+        // Make sure that all fixtures end up sharing a target dir, and we don't
+        // recompile wasm-bindgen and friends many times over.
+        static SET_TARGET_DIR: Once = ONCE_INIT;
+        SET_TARGET_DIR.call_once(|| {
+            env::set_var(
+                "CARGO_TARGET_DIR",
```
Comment (alexcrichton) [inline tests/all/utils/fixture.rs]: I'd personally recommend just always setting this env var for subprocesses, but I think `Once` will probably do the trick as well. I'm always just sort of weirdly paranoid about changing env vars while running.

File: tests/all/utils/fixture.rs [hunk 2]
```diff
@@ -87,14 +144,29 @@ impl Fixture {
...
+        assert!(shared_wasm_bindgen_test_runner.is_file());
+
+        wasm_pack::binaries::ensure_local_bin_dir(&self.path).unwrap();
+
+        fs::copy(
+            &shared_wasm_bindgen,
+            wasm_pack::binaries::local_bin_path(&self.path, "wasm-bindgen"),
+        ).expect("could not copy `wasm-bindgen` to fixture directory");
+
+        fs::copy(
```
Comment (alexcrichton) [inline tests/all/utils/fixture.rs]: For an even speedier test I'd recommend using `hard_link` first and falling back to `copy` only if that fails

File: tests/all/utils/fixture.rs (hunk 1 above)
Comment (fitzgen) [reply tests/all/utils/fixture.rs]: FWIW, this part is just code motion and we were already doing this.

**Comment by alexcrichton** [comment]
Seems fine by me!

I personally like having the fixtures next to the test itself (so you don't have to jump around too much), but that may just be me and also wouldn't want to cause too much churn for this!

File: tests/all/utils/fixture.rs (hunk 1 above)
Comment (alexcrichton) [reply tests/all/utils/fixture.rs]: 👍 no worries, feel free to move on

Comment (fitzgen) [reply tests/all/utils/fixture.rs]: (Also, the actual subprocess spawning happens deep within wasm-pack, not in the tests :-/ I agree with the general principal, however!)

File: tests/all/utils/fixture.rs (hunk 2 above)
Comment (fitzgen) [reply tests/all/utils/fixture.rs]: done!

**Pull request merged**